File: 0101.对称二叉树/0101-对称二叉树.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def is_mirror(n1, n2):
            if not n1 and not n2:
                return True
            if not n1 or not n2:
                return False
            return n1.val == n2.val and \
                    is_mirror(n1.left, n2.right) and \
                    is_mirror(n1.right, n2.left)
        
        return is_mirror(root.left, root.right) if root else True
        
                    
        

File: 0101.对称二叉树/0101-对称二叉树.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    function help(n1, n2) {
        if (n1 === null && n2 === null) {
            return true;
        }
        if (n1 === null || n2 === null) {
            return false;
        }
        return n1.val === n2.val && help(n1.left, n2.right) && help(n1.right, n2.left);
    }
    
    return root === null? true: help(root.left, root.right);
};